pkg/errors/factory.go

```diff
@@ -52,15 +59,53 @@ func newStatusError(err error) error {
 	if errAPIStatus.Status().Details == nil {
 		return err
 	}
+	canFindResource := "unknown"
 	var knerr *KNError
 	if isCRDError(errAPIStatus) {
-		knerr = NewInvalidCRD(errAPIStatus.Status().Details.Group)
+		if strings.Contains(errAPIStatus.Status().Message, "the server could not find") {
+			resourceName := getResourceNameFromErrMessage(errAPIStatus.Status().Message)
+			var kubeconfig *string
+			if home := homedir.HomeDir(); home != "" {
+				kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+			} else {
+				kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+			}
+			flag.Parse()
+			// use the current context in kubeconfig
+			config, _ := clientcmd.BuildConfigFromFlags("", *kubeconfig)
```

Comment:
I'm not a huge fan of initializing new kube client instance in such a manner. We carry-over `KnParams` instance that holds all the clients commonly used, in addition it covers Kube's ClientConfig parsing, reflecting `--kubeconfig`  and `--context` flags.

I'd rather inject the KnParams into the new error functions and retrieve the needed clients from it.